packages/stockflux-core/src/custom-hooks/useChildWindow/reducers/ChildWindow.js

```diff
@@ -0,0 +1,26 @@
+import childWindowState from './State';
+import actionType from './Actions';
+
+export const initialChildWindowState = {
+  window: null,
+  state: childWindowState.initial
+};
+
+export default (state, action) => {
+  switch (action.type) {
+    case actionType.changeState:
+      if (!Object.keys(childWindowState).indexOf(action.payload))
```

Comment:
Is this likely to occur or is it more a sanity check?

Reply:
This is checking if a valid state has been passed in, aka if `action.payload` state is one of `childWindowState`. It is unlikely to occur, but I think it's better to handle a case where we pass in an invalid state that to assume that it always comes in valid.

Reply:
I am not sure, the childWindow is a copy of the existing one, which I have moved from stockflux-launcher to stockflux-core. Eventually this will be replaced by the open-find-react-hooks library once the changes are merged in #1296.